### PR TITLE
feat: Add storage layout check script to ci 

### PR DESCRIPTION
**Pull request by nicholaspai**
Simple script that errors if developer makes changes that would change the storage layout but hasn't committed new storage layout file

### CODE REVIEW

**Review by james-a-morris — COMMENTED** [review]
I can put up a test PR that changes the diff to test that this fails. Otherwise LGTM

File: scripts/checkStorageLayout.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    done
+    exit 0
+fi
+
+for CONTRACT in "${CONTRACTS[@]}"; do
+    echo "Comparing storage layout snapshot of the $CONTRACT contract at ./storage-layouts/$CONTRACT.json with current storage layout"
+    touch ./storage-layouts/proposed.$CONTRACT.json
+    echo "Created temporary storage layout file at ./storage-layouts/proposed.$CONTRACT.json"
+    forge inspect $CONTRACT storagelayout > ./storage-layouts/proposed.$CONTRACT.json
+    echo "'forge inspect' command created temp. storage lay out file!"
```
Comment (mrice32) [inline scripts/checkStorageLayout.sh:18]:
```suggestion
    echo "'forge inspect' command created temp storage layout file!"
```

File: scripts/checkStorageLayout.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    ## TODO: add an automatic check here like, if there is diff, throw error. This forces
+    ## the developer to include the updated JSON file in the storage-layouts/ directory
+    ## in their commit, assuming this script is running in CI.
```
Comment (mrice32) [inline scripts/checkStorageLayout.sh:21]: Wait, isn't this what the if statement below does? Or am I misunderstanding this TODO?

File: scripts/diffStorageLayout.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+fi
+
+echo "Creating a storage layout snapshot of the $CONTRACT contract at ./storage-layouts/proposed.$CONTRACT.json"
+echo "⏳ This can take a few minutes to complete if you haven't run a forge command recently"
+touch ./storage-layouts/proposed.$CONTRACT.json
+
+forge inspect $CONTRACT storagelayout > ./storage-layouts/proposed.$CONTRACT.json
+echo "✅ Done!"
+vim -d ./storage-layouts/proposed.$CONTRACT.json ./storage-layouts/$CONTRACT.json
+;;
```
Comment (mrice32) [inline scripts/diffStorageLayout.sh:17]: Why the double semicolon?

File: scripts/checkStorageLayout.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+CONTRACTS=("Arbitrum_SpokePool" "Optimism_SpokePool" "Polygon_SpokePool" "Linea_SpokePool" "ZkSync_SpokePool" "Ethereum_SpokePool" "Base_SpokePool")
+if [[ "$1" == "--overwrite" ]]; then
```
Comment (mrice32) [inline scripts/checkStorageLayout.sh:4]: +1 this will be very convenient

**Review by mrice32 — APPROVED** [review]
Looks great! Just minor nits and questions about comments

File: scripts/diffStorageLayout.sh [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+CONTRACT=$1
+if [ -z "$CONTRACT" ]
+then
+  echo "Error: CONTRACT should be name of contract that you want to inspect storage layout of. e.g. Arbitrum_SpokePool"
+  exit 1
+fi
+
+echo "Creating a storage layout snapshot of the $CONTRACT contract at ./storage-layouts/proposed.$CONTRACT.json"
+echo "⏳ This can take a few minutes to complete if you haven't run a forge command recently"
+touch ./storage-layouts/proposed.$CONTRACT.json
```
Comment (rkokhatskyi) [inline scripts/diffStorageLayout.sh:12]: Hi @nicholaspai, you don't need to create a file here as you redirect the output from Foundry storage inspect to the file (line 14). If the inspect command fails, the empty file will be created anyway

File: .github/workflows/pr.yml [hunk 6]
```diff
@@ -39,6 +39,28 @@ jobs:
         run: yarn build
       - name: Test
         run: yarn test
+  forge:
+    name: Forge
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: [16]
```
Comment (rkokhatskyi) [inline .github/workflows/pr.yml:47]: Is there a need to use matrix? Looks like it is only one param to use (not sure if you guys are planning to run with other nodes too). By the way node 16 is no longer maintained and considered deprecated https://nodejs.org/en/blog/announcements/nodejs16-eol

File: scripts/checkStorageLayout.sh (hunk 2 above)
Comment (nicholaspai) [reply scripts/checkStorageLayout.sh:21]: yeah it does this is left over

File: scripts/diffStorageLayout.sh (hunk 3 above)
Comment (nicholaspai) [reply scripts/diffStorageLayout.sh:17]: idk, copilot wrote this one

File: scripts/diffStorageLayout.sh (hunk 5 above)
Comment (nicholaspai) [reply scripts/diffStorageLayout.sh:12]: nice find, thanks!

File: .github/workflows/pr.yml (hunk 6 above)
Comment (nicholaspai) [reply .github/workflows/pr.yml:47]: yeah we need to move off of this but our stack currently can't automatically upgrade. @james-a-morris is working on this

**Pull request merged**